doc/memos/rdm4.md

```diff
@@ -0,0 +1,87 @@
+RIOT Maintainers Community
+Developer Memo: 4
+Author(s): Martine Lenders
+Date: January 2017
+
+Writing tests
+=============
+
+## Abstract
+This memo describes how to write tests for RIOT.
+
+
+## Status
+This document is a product of the community of RIOT maintainers, and aims to 
+represent the consensus within this community.
+The content of this document is licensed with a Creative Commons CC-BY-SA 
+license.
+
+## Terminology
+This memo uses the following terminology and acronyms:
+
+- embUnit: a unittest framework for C that is used by RIOT to write unittests.
+- pexpect: a python library to interact with the stdio of command-line
+  application.
+- CI system: CI = Continuous Intergration, platform for automated building and
+  testing of provided changes.
+
+## 1. Introduction
+At the moment of writing there are two strategies for testing: providing a
+unittest suite in [`tests/unittests/`][unittests] or a stand-alone test
+application in `tests/` directly. This document provides some recommendations
+on where to put a test and what the minimum requirements for those are.
+
+## 2. Unittests
+Unittests in RIOT are provided using a costumized fork of the [embUnit]
+framework. While it is possible to provide a stand-alone application using this
+framework it is recommended to add or extend a test suite in the [unittests]
+application (see [unittests README] for how to do that).
+
+The advantage of that is that this application is at the time of writing this
```

Comment:
some words too much. I'd drop the description of the current state, as that will hopefully change soon. No need to write things that will be obsolete in the short term.